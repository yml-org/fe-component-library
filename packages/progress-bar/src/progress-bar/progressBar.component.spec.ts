import { LitElement } from 'lit-element';

describe('progress-bar-component', () => {
  const PROGRESS_BAR_COMPONENT = 'progress-bar-component';
  let progressBarEle: LitElement;

  const getShadowRoot = (tagName: string): ShadowRoot =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    progressBarEle = window.document.createElement(
      PROGRESS_BAR_COMPONENT
    ) as LitElement;
    document.body.appendChild(progressBarEle);
  });

  afterEach(() => {
    document.body.getElementsByTagName(PROGRESS_BAR_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(PROGRESS_BAR_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('renders with default values', () => {
    const progressBar = getShadowRoot(PROGRESS_BAR_COMPONENT).querySelector(
      '[part="webcl-progress-bar"]'
    );
    expect(progressBar).toBeDefined();
    expect(progressBar.getAttribute('aria-valuemin')).toEqual('0');
    expect(progressBar.getAttribute('aria-valuemax')).toEqual('100');
  });

  it('renders with custom values', async () => {
    progressBarEle['percent'] = 50;
    progressBarEle['max'] = 200;
    await progressBarEle.updateComplete;
    const progressBar = getShadowRoot(PROGRESS_BAR_COMPONENT).querySelector(
      '[part="webcl-progress-bar"]'
    );
    expect(progressBar.getAttribute('aria-valuemin')).toEqual('0');
    expect(progressBar.getAttribute('aria-valuemax')).toEqual('200');
    expect(progressBar.getAttribute('aria-valuenow')).toEqual('50');
  });

  it('updates the progress bar on property change', async () => {
    const progressBar = getShadowRoot(PROGRESS_BAR_COMPONENT).querySelector(
      '[part="webcl-progress-bar"]'
    );
    progressBarEle['percent'] = 75;
    await progressBarEle.updateComplete;
    expect(progressBar.getAttribute('aria-valuenow')).toEqual('75');
  });
});
