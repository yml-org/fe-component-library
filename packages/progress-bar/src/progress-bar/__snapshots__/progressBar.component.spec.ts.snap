// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress-bar-component matches component snapshot 1`] = `
H {
  "_$AL": Map {},
  "_$Do": C {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": mt {
      "_$AD": k {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 1,
            "name": "style",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 1,
            "name": "aria-valuemax",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 1,
            "name": "aria-valuenow",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        T {
          "_$AH": "width: NaN%; transition: width 0.3s ease-in-out;",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow=""
            part="webcl-progress-bar"
            role="progressbar"
            style="width: NaN%; transition: width 0.3s ease-in-out;"
          />,
          "name": "style",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        T {
          "_$AH": 100,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow=""
            part="webcl-progress-bar"
            role="progressbar"
            style="width: NaN%; transition: width 0.3s ease-in-out;"
          />,
          "name": "aria-valuemax",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        T {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow=""
            part="webcl-progress-bar"
            role="progressbar"
            style="width: NaN%; transition: width 0.3s ease-in-out;"
          />,
          "name": "aria-valuenow",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__max": 100,
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": C {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": mt {
        "_$AD": k {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 1,
              "name": "style",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 1,
              "name": "aria-valuemax",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 1,
              "name": "aria-valuenow",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          T {
            "_$AH": "width: NaN%; transition: width 0.3s ease-in-out;",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow=""
              part="webcl-progress-bar"
              role="progressbar"
              style="width: NaN%; transition: width 0.3s ease-in-out;"
            />,
            "name": "style",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          T {
            "_$AH": 100,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow=""
              part="webcl-progress-bar"
              role="progressbar"
              style="width: NaN%; transition: width 0.3s ease-in-out;"
            />,
            "name": "aria-valuemax",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          T {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow=""
              part="webcl-progress-bar"
              role="progressbar"
              style="width: NaN%; transition: width 0.3s ease-in-out;"
            />,
            "name": "aria-valuenow",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
